### DISCUSSION TITLE
"using" key in code when decompile

### DISCUSSION THREAD

**Discussion by yanivchik** [icsharpcode/ILSpy#3193] [Development]
Hi

I have a question:

if I have original code like that (just example):

```
using( var context = new Contex())
{
  var test = context.fetch();
}
```

when after I use ILSpy, I'm getting this

```
Contex context = new Contex();
try
{
  var test = context.fetch();
}
finally
{
  ((IDisposable)sochenEntities)?.Dispose();
}
```

it something that ILSpy can not recognize "using" context and translate it into try-catch with dispose block?

(using .NET framework 4.6.2)

thank you

**Comment by siegfriedpammer** [comment]
Can you please provide an example binary for me to check? Thanks!

**Reply by siegfriedpammer** [reply]
In your example the variable used in the finally block is not the same as the using variable. Is this the actual code or a mistake? If it's the actual code, then that's the reason.
